app/src/main/java/com/douglasbruce/tasky/features/register/RegisterScreen.kt

```diff
@@ -109,10 +101,20 @@ internal fun RegisterScreen(
                     modifier = Modifier.fillMaxSize()
                 ) {
                     TaskyTextField(
-                        value = name,
-                        onValueChange = { name = it },
-                        isValid = true,
-                        placeholder = R.string.name_placeholder,
+                        value = uiState.name,
+                        onValueChange = {
+                            onEvent(RegistrationFormEvent.NameValueChanged(it))
+                        },
+                        isError = uiState.nameErrorType != ErrorType.NONE,
+                        isValid = uiState.isNameValid,
+                        placeholder = stringResource(R.string.name_placeholder),
+                        supportingText = when (uiState.nameErrorType) {
+                            ErrorType.LENGTH -> {
+                                { Text(text = stringResource(id = R.string.name_error_length)) }
+                            }
+
+                            else -> null
+                        },
```

Comment:
Could it make more sense here to move the `when` expression inside the lambda, so you only need one lambda and don't end up with so many curly brackets? Or does that lead to some undesired behavior if there's no supporting text and the lambda is non-null?

Reply:
Because supportingText is non-null it adds extra spacing

![image](https://github.com/DouglasBruce/Tasky/assets/15791449/469b2e44-ec7b-497c-acee-f096b90c6537)

I know I could workaround it and still get the desired layout, but I feel like this would cause some confusion to others, especially if they weren't aware of the default styling of the supportingText.

Reply:
Alright, makes sense, thanks!